src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
This a fix value, you can move it to the field declaration (as it was before)

````java
protected final Iterable<Text> childrenIterable = () ->
        this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
````
or change it check for `isEmpty()` before it is set.
````java
if (this.children.isEmpty()) {
    this.childrenIterable = () -> Iterators.singletonIterator(this);
} else {
    this.childrenIterable = new TextIterator(this);
}
````